src/v/datalake/translation/scheduling.h

```diff
@@ -0,0 +1,250 @@
+/*
+ * Copyright 2024 Redpanda Data, Inc.
+ *
+ * Licensed as a Redpanda Enterprise file under the Redpanda Community
+ * License (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * https://github.com/redpanda-data/redpanda/blob/master/licenses/rcl.md
+ */
+#pragma once
+
+#include "config/property.h"
+#include "container/chunked_hash_map.h"
+#include "container/fragmented_vector.h"
+#include "ssx/semaphore.h"
+
+#include <seastar/core/abort_source.hh>
+#include <seastar/core/condition-variable.hh>
+#include <seastar/core/gate.hh>
+
+namespace datalake::translation::scheduling {
+
+using clock = ss::lowres_clock;
+
+using translator_id = model::ntp;
+
+/**
+ * Hooks for the components to notify the scheduler about events.
+ */
+class state_change_notifier {
```

Comment:
does this need a virtual destructor?

Reply:
Same question. I think a note about the intended ownership semantics would be helpful. I see polymorphic non-owning references at various interfaces, but at a glance it's not clear to me where the actual scheduler object will sit.

Reply:
cleaned it up a bit more, lmk if it still not clear.